README.md
# foliogallery20
foliogallery 2.0


changelog.txt
version 2.0
- Added ability to embed videos from youtube and vimeo
- Added ability to add descriptions to albums and images from a text file
- Modified underlying code to accommodate the above changes

version 1.4
- Security fixes
- minor CSS changes

version 1.3
- Added option to display random or static album thumbnails
- Modified thumbnail resizing function. Now it resizes images only if they are wider than specified
- Renamed some php variables in code
- Sort by name is not case sensitive now
- Cleaned up code and fixed some other bugs
- Changed appearance of pagination   
- Made other minor modifications

version 1.2
- Renamed the gallery to folioGallery
- Added option for displaying captions in album files
- Added png and gif automatic thumbnail creation
- Changed thumbnail styling
- Modified the CSS 
- Made other minor modifications

Version 1.1
- Correct number of images are now displayed in colorbox modal window.
- Correct number of images are now looped in colorbox modal window. 
- Made minor changes to php file.

Create folder resource
ln -s /Volumes/DATA/_Downloads/_my_albums ./_albums
ln -s /Volumes/DATA/_Downloads/_my_videos/ ./_videos

Get youtube thumb image
Link => http://img.youtube.com/vi/[Youtube Code]/[number - 1,2,3].jpg
http://www.reelseo.com/youtube-thumbnail-image/

index.php
<?php

define("__USERNAME", "admin");
define("__PASSWORD", "admin");

if (!isset($_SERVER['PHP_AUTH_USER']))
{
    header("WWW-Authenticate: Basic realm=\"Private Area\"");
    header("HTTP/1.0 401 Unauthorized");
    print "Sorry - you need valid credentials to be granted access!\n";
    exit;
} else
{
    if (($_SERVER['PHP_AUTH_USER'] != __USERNAME) && ($_SERVER['PHP_AUTH_PW'] != __PASSWORD))
    {
        header("WWW-Authenticate: Basic realm=\"Private Area\"");
        header("HTTP/1.0 401 Unauthorized");
        print "Sorry - you need valid credentials to be granted access!\n";
        exit;
    }

}

require_once 'foliogallery/mobile_detect.php';

$detect = new Mobile_Detect;
$videoFrameWidth = "800";
$videoFrameHeight = "600";

// Any tablet device.
if( $detect->isTablet() )
{
    $videoFrameWidth = "600";
    $videoFrameHeight = "450";
}


// Any mobile device (phones or tablets).
if ( $detect->isMobile() )
{
    $videoFrameWidth = "400";
    $videoFrameHeight = "300";
}


?>

<!DOCTYPE HTML>
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<title>Image Gallery By FolioPages.com</title>
<style type="text/css">
body {
background:#eee;
margin:0;
padding:0;
font:12px arial, Helvetica, sans-serif;
color:#222;
}
</style>
<link type="text/css" rel="stylesheet" href="foliogallery/foliogallery.css" />
<link type="text/css" rel="stylesheet" href="colorbox/colorbox.css" />
<script type="text/javascript" src="foliogallery/jquery.1.11.js"></script>
<script type="text/javascript" src="colorbox/jquery.colorbox-min.js"></script>
<script type="text/javascript" src="foliogallery/foliogallery.js"></script>


<script type="text/javascript">
$(document).ready(function(){
    // initiate colorbox
	$('.albumpix').colorbox({rel:'albumpix', maxWidth:'96%', maxHeight:'96%', slideshow:true, slideshowSpeed:3500, slideshowAuto:false});
	$('.vid').colorbox({rel:'albumpix', iframe:true, width:'80%', height:'96%'});

    $(".flowplayer").colorbox({rel: 'nofollow', iframe:true, scrolling: false, innerWidth:'<?php echo $videoFrameWidth; ?>', innerHeight:'<?php echo $videoFrameHeight; ?>'});
});
</script>
</head>
<body>

<br />
<br />

<?php $_REQUEST['fullalbum']=1; include 'foliogallery/foliogallery.php'; ?>

<br />
<br />


</body>
</html>

foliogallery-demo.php
<!DOCTYPE html>
<head>
<title>Image Gallery By FolioPages.com</title>
<style type="text/css">
body {
background:#eee;
margin:0;
padding:0;
font:12px arial, Helvetica, sans-serif;
color:#222;
}
</style>
<link type="text/css" rel="stylesheet" href="foliogallery/foliogallery.css" />
<link type="text/css" rel="stylesheet" href="colorbox/colorbox.css" />
<script type="text/javascript" src="foliogallery/jquery.1.11.js"></script>
<script type="text/javascript" src="foliogallery/foliogallery.js"></script>
<script type="text/javascript" src="colorbox/jquery.colorbox-min.js"></script>
</head>
<body>

<br />
<br />

<!-- display images located in "Big Bear California" subfolder and paginate in 3 item blocks -->
<div id="folioGallery1" class="folioGallery" title="Big Bear California"><div class="numPerPage" title="3"></div></div>

<br />

<!-- display images located in "Las Vegas Strip" subfolder and paginate in 3 item blocks -->
<div id="folioGallery3" class="folioGallery" title="Mixed Content"><div class="numPerPage" title="3"></div></div>

<br />
<br />

<!-- display list of folders in "albums" folder and paginate in 6 item blocks -->
<div id="folioGallery" class="folioGallery"><div class="numPerPage" title="3"></div></div>

<br />
<br />

<div align="center">folioGallery - Installation and instructions @ <a href="http://foliopages.com/php-jquery-ajax-photo-gallery-no-database">FolioPages.com</a></div>

<br />
<br />


</body>
</html>

foliogallery-demo2.php
<?php

define("__USERNAME", "admin");
define("__PASSWORD", "admin");

//if (!isset($_SERVER['PHP_AUTH_USER']))
//{
//    header("WWW-Authenticate: Basic realm=\"Private Area\"");
//    header("HTTP/1.0 401 Unauthorized");
//    print "Sorry - you need valid credentials to be granted access!\n";
//    exit;
//} else
//{
//    if (($_SERVER['PHP_AUTH_USER'] != __USERNAME) && ($_SERVER['PHP_AUTH_PW'] != __PASSWORD))
//    {
//        header("WWW-Authenticate: Basic realm=\"Private Area\"");
//        header("HTTP/1.0 401 Unauthorized");
//        print "Sorry - you need valid credentials to be granted access!\n";
//        exit;
//    }
//
//}

require_once 'mobile_detect.php';

$detect = new Mobile_Detect;
$videoFrameWidth = "800";
$videoFrameHeight = "600";

// Any tablet device.
if( $detect->isTablet() )
{
    $videoFrameWidth = "600";
    $videoFrameHeight = "450";
}


// Any mobile device (phones or tablets).
if ( $detect->isMobile() )
{
    $videoFrameWidth = "400";
    $videoFrameHeight = "300";
}


?>

<!DOCTYPE HTML>
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<title>Image Gallery By FolioPages.com</title>
<style type="text/css">
body {
background:#eee;
margin:0;
padding:0;
font:12px arial, Helvetica, sans-serif;
color:#222;
}
</style>
<link type="text/css" rel="stylesheet" href="foliogallery/foliogallery.css" />
<link type="text/css" rel="stylesheet" href="colorbox/colorbox.css" />
<script type="text/javascript" src="foliogallery/jquery.1.11.js"></script>
<script type="text/javascript" src="colorbox/jquery.colorbox-min.js"></script>
<script type="text/javascript">
$(document).ready(function(){
    // initiate colorbox
	$('.albumpix').colorbox({rel:'albumpix', maxWidth:'96%', maxHeight:'96%', slideshow:true, slideshowSpeed:3500, slideshowAuto:false});
	$('.vid').colorbox({rel:'albumpix', iframe:true, width:'80%', height:'96%'});

    $(".flowplayer").colorbox({rel: 'nofollow', iframe:true, scrolling: false, innerWidth:'<?php echo $videoFrameWidth; ?>', innerHeight:'<?php echo $videoFrameHeight; ?>'});
});
</script>
</head>
<body>

<br />
<br />

<?php $_REQUEST['fullalbum']=1; include 'foliogallery.php'; ?>

<br />
<br />

<!--<div align="center">folioGallery - Installation and instructions @ <a href="http://foliopages.com/php-jquery-ajax-photo-gallery-no-database">FolioPages.com</a></div>-->

<br />
<br />


</body>
</html>

foliogallery/foliogallery.js
/*!
	folioGallery v1.4 - 2015-01-30
	(c) 2014 Harry Ghazanian - foliopages.com/php-jquery-ajax-photo-gallery-no-database
	This content is released under the http://www.opensource.org/licenses/mit-license.php MIT License.
*/
$(function() {		
				
	var folioGalleryDir = './foliogallery'; // foliogallery folder relative path - absolute path like http://my_website.com/foliogallery may not work
		
	// find divs with class folioGallery and load album in it based on its id
	$('.folioGallery').each(function() {
		
		var targetDiv = (this.id); // id of div to load albums
		var numPerPage = $('#'+targetDiv+' div.numPerPage').prop('title');
								
		if(targetDiv=='folioGallery') {
			var fullAlbum = 1;
			var showAlb = ''; // empty will show full gallery
		} else {
			var fullAlbum = 0;
			var showAlb = $('#'+targetDiv).prop('title'); // title attribute of div - same as album folder
		}
		
		loadGallery(folioGalleryDir,targetDiv,showAlb,numPerPage,1,fullAlbum); // inital load
				
		// in gallery view, load album when thumb is clicked
		$(this).on('click', 'a.showAlb', function() {	
			var showAlb = $(this).prop('rel');
			loadGallery(folioGalleryDir,targetDiv,showAlb,numPerPage,1,fullAlbum);
			return false;
		});	
				
		// paginate albums and pics
		$(this).on('click', 'a.pag', function() {

            console.log('show here');
			var showAlb = $(this).prop('rel');
			var pageNum = $(this).prop('rev');
			loadGallery(folioGalleryDir,targetDiv,showAlb,numPerPage,pageNum,fullAlbum);
			return false;
		});
		
		// refresh div content
		$(this).on('click', 'a.refresh', function() {
		   loadGallery(folioGalleryDir,targetDiv,'',numPerPage,1,fullAlbum);
		   return false;
		});
				
	});
		
});

function loadGallery(folioGallerydir,targetdiv,album,numperpage,pagenum,fullalbum) {                    
	$.ajax
	({
		type: 'POST',
		url: folioGallerydir+'/foliogallery.php?'+album+'&p='+pagenum,
		data: {
			album: album,
			numperpage: numperpage,
			pagenum: pagenum,
			fullalbum: fullalbum
		},
		cache: false,
		success: function(msg)
		{
			$('#'+targetdiv).html(msg).hide().show();
		}
	});
	return false;
}

foliogallery/show_video_html5.php
<?php 

$video_file = !empty($_REQUEST['video']) ? $_REQUEST['video']: exit('No direct script access allowed');
$innerWith = !empty($_REQUEST['innerWith']) ? $_REQUEST['innerWith']: "800";
$innerHeight = !empty($_REQUEST['innerHeight']) ? $_REQUEST['innerHeight']: "600";

$ext = strtolower(pathinfo($_REQUEST['video'], PATHINFO_EXTENSION));
$video_type = "";
switch ($ext)
{
	case "mp4":
		$video_type = "video/mp4";
		break;
	case "webm":
		$video_type = "video/webm";
		break;
	case "flv":
		$video_type = "video/x-flv";
		break;
}



?>

<!DOCTYPE HTML>
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<title></title>

<style type="text/css">
body {
background:#eee;
margin:0;
padding:0;
font:12px arial, Helvetica, sans-serif;
color:#222;
}
</style>
</head>
<body>

<video width="<?php echo $innerWith; ?>" height="<?php echo $innerHeight; ?>" controls autoplay>
  	<source src="VideoStream/Stream.php?video=<?php echo $video_file; ?>" type="<?php echo $video_type; ?>">
	Your browser does not support the video tag.
</video>

</body>
</html>

foliogallery/VideoStream/VideoStream.php
<?php
/**
 * Description of VideoStream
 *
 * @author Rana
 * @link http://codesamplez.com/programming/php-html5-video-streaming-tutorial
 */
class VideoStream
{
    private $path = "";
    private $stream = "";
    private $buffer = 102400;
    private $start  = -1;
    private $end    = -1;
    private $size   = 0;
 
    function __construct($filePath) 
    {
        $this->path = $filePath;
    }
     
    /**
     * Open stream
     */
    private function open()
    {
        if (!($this->stream = fopen($this->path, 'rb'))) {
            die('Could not open stream for reading');
        }
         
    }
     
    /**
     * Set proper header to serve the video content
     */
    private function setHeader()
    {
        ob_get_clean();
        header("Content-Type: video/mp4");
        
        $ext = strtolower(pathinfo($this->path, PATHINFO_EXTENSION));
        switch ($ext)
        {
        	case "mp4":
        		header("Content-Type: video/mp4");
        		break;
        	case "webm":
        		header("Content-Type: video/webm");
        		break;
        	case "flv":
        		header("Content-Type: video/x-flv");
        		break;        		
        }        
        
        header("Cache-Control: max-age=2592000, public");
        header("Expires: ".gmdate('D, d M Y H:i:s', time()+2592000) . ' GMT');
        header("Last-Modified: ".gmdate('D, d M Y H:i:s', @filemtime($this->path)) . ' GMT' );
        $this->start = 0;
        $this->size  = filesize($this->path);
        $this->end   = $this->size - 1;
        header("Accept-Ranges: 0-".$this->end);
         
        if (isset($_SERVER['HTTP_RANGE'])) {
  
            $c_start = $this->start;
            $c_end = $this->end;
 
            list(, $range) = explode('=', $_SERVER['HTTP_RANGE'], 2);
            if (strpos($range, ',') !== false) {
                header('HTTP/1.1 416 Requested Range Not Satisfiable');
                header("Content-Range: bytes $this->start-$this->end/$this->size");
                exit;
            }
            if ($range == '-') {
                $c_start = $this->size - substr($range, 1);
            }else{
                $range = explode('-', $range);
                $c_start = $range[0];
                 
                $c_end = (isset($range[1]) && is_numeric($range[1])) ? $range[1] : $c_end;
            }
            $c_end = ($c_end > $this->end) ? $this->end : $c_end;
            if ($c_start > $c_end || $c_start > $this->size - 1 || $c_end >= $this->size) {
                header('HTTP/1.1 416 Requested Range Not Satisfiable');
                header("Content-Range: bytes $this->start-$this->end/$this->size");
                exit;
            }
            $this->start = $c_start;
            $this->end = $c_end;
            $length = $this->end - $this->start + 1;
            fseek($this->stream, $this->start);
            header('HTTP/1.1 206 Partial Content');
            header("Content-Length: ".$length);
            header("Content-Range: bytes $this->start-$this->end/".$this->size);
        }
        else
        {
            header("Content-Length: ".$this->size);
        }  
         
    }
    
    /**
     * close curretly opened stream
     */
    private function end()
    {
        fclose($this->stream);
        exit;
    }
     
    /**
     * perform the streaming of calculated range
     */
    private function stream()
    {
        $i = $this->start;
        set_time_limit(0);
        while(!feof($this->stream) && $i <= $this->end) {
            $bytesToRead = $this->buffer;
            if(($i+$bytesToRead) > $this->end) {
                $bytesToRead = $this->end - $i + 1;
            }
            $data = fread($this->stream, $bytesToRead);
            echo $data;
            flush();
            $i += $bytesToRead;
        }
    }
     
    /**
     * Start streaming video content
     */
    function start()
    {
        $this->open();
        $this->setHeader();
        $this->stream();
        $this->end();
    }
}

foliogallery/foliogallery.php
<!--
	folioGallery v2.0 - 2015-01-30
	(c) 2014 Harry Ghazanian - foliopages.com/php-jquery-ajax-photo-gallery-no-database
	This content is released under the http://www.opensource.org/licenses/mit-license.php MIT License.
-->
<script type="application/javascript">
$(document).ready(function() { // load colorbox settings
	$('.folioGallery').each(function() {
		var targetDiv = (this.id); // id of div to load albums	
		$('#'+targetDiv+' .albumpix').colorbox({rel:targetDiv, maxWidth:'96%', maxHeight:'96%', slideshow:true, slideshowSpeed:3500, slideshowAuto:false});
		$('#'+targetDiv+' .vid').colorbox({rel:targetDiv, iframe:true, width:"80%", height:"96%"});
	});
});
</script>
<?php
/***** gallery settings *****/
$mainFolder          = '_albums'; // main folder that holds albums - this folder resides on root directory of your domain
$album_page_url      = $_SERVER['PHP_SELF']; // url of page where gallery/albums are located 
$no_thumb            = 'foliogallery/noimg.png';  // show this when no thumbnail exists 
$extensions          = array("jpg","png","gif","JPG","PNG","GIF"); // allowed extensions in photo gallery 
$itemsPerPage        = '16';    // number of images per page if not already specified in ajax mode
$thumb_width         = '150';   // width of thumbnails in pixels
$sort_albums_by_date = TRUE;    // TRUE will sort albums by creation date (most recent first), FALSE will sort albums by name 
$sort_images_by_date = TRUE;    // TRUE will sort thumbs by creation date (most recent first), FALSE will sort images by name 
$random_thumbs       = TRUE;    // TRUE will display random thumbnails, FALSE will display the first image from thumbs folders
$show_captions       = TRUE;    // TRUE will display file names as captions on thumbs inside albums, FALSE will display no captions
$num_captions_chars  = '20';    // number of characters displayed in album and thumb captions

$mainVideo          = '_videos'; // main folder that holds albums - this folder resides on root directory of your domain
$extensionsVideo    = array("avi","mp4","wmv"); // allowed extensions in video gallery
$num_video_captions_chars  = '30';    // number of characters displayed in album and thumb captions

/***** end gallery settings *****/

$numPerPage = (!empty($_REQUEST['numperpage']) ? (int)$_REQUEST['numperpage'] : $itemsPerPage);
$fullAlbum  = (!empty($_REQUEST['fullalbum']) ? 1 : 0);
 
// sanitize string
function sanitize($string)
{
	$string = htmlspecialchars(trim($string), ENT_QUOTES, 'UTF-8');
	return $string;
}

// encode string to
function encodeto($string)
{
	$string = mb_convert_encoding(trim($string), 'UTF-8', 'HTML-ENTITIES');
	return $string;
} 
 
// function to create thumbnails from images
function make_thumb($folder,$file,$dest,$thumb_width) {

	$ext = strtolower(pathinfo($file, PATHINFO_EXTENSION));
	
	switch($ext)
	{
		case "jpg":
		$source_image = imagecreatefromjpeg($folder.'/'.$file);
		break;
		
		case "jpeg":
		$source_image = imagecreatefromjpeg($folder.'/'.$file);
		break;
		
		case "png":
		$source_image = imagecreatefrompng($folder.'/'.$file);
		break;
		
		case "gif":
		$source_image = imagecreatefromgif($folder.'/'.$file);
		break;
	}	
	
	$width = imagesx($source_image);
	$height = imagesy($source_image);
	
	if($width < $thumb_width) // if original image is smaller don't resize it
	{
		$thumb_width = $width;
		$thumb_height = $height;
	}
	else
	{
		$thumb_height = floor($height*($thumb_width/$width));
	}
	
	$virtual_image = imagecreatetruecolor($thumb_width,$thumb_height);
	
	if($ext == "gif" or $ext == "png") // preserve transparency
	{
		imagecolortransparent($virtual_image, imagecolorallocatealpha($virtual_image, 0, 0, 0, 127));
		imagealphablending($virtual_image, false);
		imagesavealpha($virtual_image, true);
    }
	
	imagecopyresampled($virtual_image,$source_image,0,0,0,0,$thumb_width,$thumb_height,$width,$height);
	
	switch($ext)
	{
	    case 'jpg': imagejpeg($virtual_image, $dest,80); break;
		case 'jpeg': imagejpeg($virtual_image, $dest,80); break;
		case 'gif': imagegif($virtual_image, $dest); break;
		case 'png': imagepng($virtual_image, $dest); break;
    }

	imagedestroy($virtual_image); 
	imagedestroy($source_image);
	
}

// return array sorted by date or name
function sort_array(&$array,$dir,$sort_by_date) { // array argument must be passed as reference
	
	if($sort_by_date)
	{
		foreach ($array as $key=>$item) 
		{
			$stat_items = stat($dir .'/'. $item);
			$item_time[$key] = $stat_items['ctime'];
		}
		return array_multisort($item_time, SORT_DESC, $array); 
	}	
	else
	{
		return usort($array, 'strnatcasecmp');
	}	

}

// get album and image descriptions
function itemDescription($album, $file='')
{
	if(file_exists($album.'/descriptions.txt'))
	{
		$lines_array = file($album.'/descriptions.txt', FILE_IGNORE_NEW_LINES | FILE_SKIP_EMPTY_LINES); 
		if($lines_array)
		{
			if($file == '')
			{
				$album_line = explode(';', $lines_array[0]); 
				return (!empty($album_line[0]) && $album_line[0] == 'album' ? $album_line[1] : '');
			}
			else
			{
				foreach($lines_array as $img_line)
				{	
					if(!empty($img_line)) {
						$img_desc = explode(';', $img_line);	
						if($img_desc[0] == $file) { return $img_desc[1]; }
					}
				}
			}	
		}
		else
		{
			return '';
		}
	}	
}


// display pagination
function paginate_array($numPages,$urlVars,$alb,$currentPage) {
        
   $html = '';
   
   if ($numPages > 1) 
   {
   
       if ($currentPage > 1)
	   {
	       $prevPage = $currentPage - 1;
	       $html .= '<a class="pag prev" rel="'.$alb.'" rev="'.$prevPage.'" href="?'.$urlVars.'p='.$prevPage.'"></a> ';
	   }	   
	   
	   for( $i=0; $i < $numPages; $i++ )
	   {
           $p = $i + 1;
		   $class = ($p==$currentPage ? 'current-paginate' : 'paginate'); 
		   $html .= '<a rel="'.$alb.'" rev="'.$p.'" class="'.$class.' pag" href="?'.$urlVars.'p='.$p.'"></a>';	  
	   }
	   
	   if ($currentPage != $numPages)
	   {
           $nextPage = $currentPage + 1;	
		   $html .= ' <a class="pag next" rel="'.$alb.'" rev="'.$nextPage.'" href="?'.$urlVars.'p='.$nextPage.'"></a>';
	   }	  	 
   
   }
   
   return $html;

}
?>

<div class="fg">

<?php
if (empty($_REQUEST['album'])) // if no album requested, show all albums
{		
    $albums = array_diff(scandir($mainFolder), array('..', '.', '.DS_Store', '.htaccess'));
	$numAlbums = count($albums);
	 
	if($numAlbums == 0) 
	{ ?>
		
		<div class="titlebar"><p>No albums posted</p></div>
    
	<?php
	}
	else
	{
		sort_array($albums,$mainFolder,$sort_albums_by_date); // rearrange array either by date or name
		$numPages = ceil( $numAlbums / $numPerPage );
		
		if(isset($_REQUEST['p']))
		 {
		 	$currentPage = ((int)$_REQUEST['p'] > $numPages ? $numPages : (int)$_REQUEST['p']); 
         } 
		 else 
		 {
		 	$currentPage=1;
         }
		
		$start = ($currentPage * $numPerPage) - $numPerPage; ?>
	     
		<div class="p10-lr">
        	<span class="title">Photo Gallery</span> - <?php echo $numAlbums; ?> albums
        </div>
	  
        <div class="clear"></div>
	  	 
		<?php 			 
	    for( $i=$start; $i<$start + $numPerPage; $i++ )
		{
	  						
			if(isset($albums[$i])) 
			{  
				$thumb_pool = glob($mainFolder.'/'.$albums[$i].'/thumbs/*{.'.implode(",", $extensions).'}', GLOB_BRACE);
				
				if (count($thumb_pool) == 0)
				{ 
					$album_thumb = $no_thumb;
				}
				else
				{	
					$album_thumb = ($random_thumbs ? $thumb_pool[array_rand($thumb_pool)] : $thumb_pool[0]); // display a random thumb or the 1st thumb					
				} ?>
			 		 			 
				<div class="thumb-wrapper">
					<div class="thumb">
					   <a class="showAlb" rel="<?php echo $albums[$i]; ?>" href="<?php echo $_SERVER['PHP_SELF']; ?>?album=<?php echo urlencode($albums[$i]); ?>">
					     <img src="<?php echo $album_thumb; ?>" alt="<?php echo $albums[$i]; ?>" /> 
					   </a>	
					</div>
					<div class="caption"><?php echo substr($albums[$i],0,$num_captions_chars); ?></div>
				</div>
	
			<?php
			}
		
		}
		?>
	      
		 <div class="clear"></div>
  
         <div align="center" class="paginate-wrapper">
        	<?php
			$urlVars = "";
			$alb = "";
            echo paginate_array($numPages,$urlVars,$alb,$currentPage);
			?>
         </div>

        <!--
            Hien thi phan video
           -->

    <?php

    $videos = array_diff(scandir($mainVideo), array('..', '.', '.DS_Store', '.htaccess'));
    $numVideos = count($videos);

    if ($numVideos <= 0) :

        ?>

        <div class="titlebar"><p>There are currently no videos</p></div>

        <?php

    else :
        sort_array($videos,$mainVideo,$sort_albums_by_date); // rearrange array either by date or name

        ?>

        <br><br><br>
        <div class="p10-lr">
            <span class="title">Video Gallery</span> - <?php echo $numVideos; ?> videos
        </div>
        <div class="clear"></div>

        <?php
        for( $i=0; $i<$numVideos; $i++ ) :
            if( !isset($videos[$i])) continue;
            ?>

            <div class="thumb-wrapper">
                <div class="thumb">
                    <a class='flowplayer' rel="<?= $videos[$i]; ?>" href="foliogallery/show_video_html5.php?video=<?php echo $mainVideo.'/'.$videos[$i]; ?>&innerWith=<?php echo $videoFrameWidth; ?>&innerHeight=<?php echo $videoFrameHeight; ?>">
                        <img alt="<?php echo $numVideos[$i]; ?>" src="foliogallery/video_thumb.png">
                    </a>
                </div>
                <div class="caption"><?php echo substr($videos[$i],0,$num_video_captions_chars); ?></div>
            </div>

        <?php endfor; ?>

        <div class="clear"></div>

        <?php

    endif;

	}

} 
else //display photos in album 
{

	$albums_in_maindir = scandir($mainFolder);	
	$requested_album = sanitize($_REQUEST['album']); // xss prevention
	
	// check requested album against directory traverse
	if (!in_array($requested_album, $albums_in_maindir)) { ?>
        <span class="title">Photo Gallery &raquo; <a href="<?php echo $album_page_url; ?>" class="refresh">Root</a></span>
		<p></p>
		<?php die('Invalid Request');
	}
	
	$album = $mainFolder.'/'.$requested_album;
	$scanned_album = scandir($album);
	
	$files = array_diff($scanned_album, array('..', '.','thumbs','descriptions.txt','.DS_Store','.htaccess'));
	$numFiles = count($files); ?>
	 
	<div class="p10-lr">
		<?php if($fullAlbum==1) { ?>
			<span class="title"><a href="<?php echo $album_page_url; ?>" class="refresh">Root</a></span>
			<span class="title">&raquo;</span>
		<?php } ?>
		<span class="title"><?php echo $requested_album; ?></span> - <?php echo $numFiles; ?> images
	</div>  
	   
	<div class="clear"></div>
	
	<?php
	if($numFiles == 0)
	{ ?>
	    
		 <div class="p10-lr"><p>There are no images in this album.</p></div>
	
	<?php
	}
	else	
	{			
		sort_array($files,$album,$sort_images_by_date); // rearrange array either by date or name
		$numPages = ceil( $numFiles / $numPerPage );
		
		if(isset($_REQUEST['p']))
		{
			$currentPage = ((int)$_REQUEST['p'] > $numPages ? $numPages : (int)$_REQUEST['p']);
		} 
		 else
		{
		 	$currentPage=1;
		}
		 			 
		$start = ($currentPage * $numPerPage) - $numPerPage;
		
		if (!is_dir($album.'/thumbs')) 
		{
			mkdir($album.'/thumbs');
			chmod($album.'/thumbs', 0777);
			//chown($album.'/thumbs', 'apache'); 
		}	 	

		for( $i=0; $i <= $numFiles; $i++ )
		{   
			if(isset($files[$i]) && is_file($album .'/'. $files[$i]))
			{   		    
				$thumb_url_end = '</a>';
				$ext = strtolower(pathinfo($files[$i], PATHINFO_EXTENSION));
				$prefix = substr($files[$i], 0, -(strlen($ext)+1));
				$prefix_begin = mb_substr($prefix, 0, 6); // 1st 6 characters of caption			
			    $full_caption = (itemDescription($album, $files[$i]) ? itemDescription($album, $files[$i]) : $prefix); // image captions
				$num_chars = strlen($full_caption);
				$caption = ($num_chars > $num_captions_chars ? substr($full_caption,0,$num_captions_chars).'...' : $full_caption);
				$caption = encodeto($caption);
					
				switch($prefix_begin)
				{
					case "utube-":
					$video_id = explode('utube-', $prefix);
					$video_id = $video_id[1];
					$thumb_url_start = '<a href="http://www.youtube.com/embed/'.$video_id.'?rel=0&amp;wmode=transparent" title="'.$caption.'" class="albumpix vid">';
					break;
					
					case "vimeo-":
					$video_id = explode('vimeo-', $prefix);
					$video_id = $video_id[1];
					$thumb_url_start = '<a href="http://player.vimeo.com/video/'.$video_id.'" title="'.$caption.'" class="albumpix vid">';
					break;
					
					default:
					$thumb_url_start = '<a href="'.$album.'/'.$files[$i].'" title="'.$caption.'" class="albumpix">';
					break;
				}
									
				if(in_array($ext, $extensions)) 
				{  				  					   
					$thumb = $album.'/thumbs/'.$files[$i];
					if (!file_exists($thumb))
					{
						make_thumb($album,$files[$i],$thumb,$thumb_width); 
					}	   
				   
					if($i<$start || $i>=$start + $numPerPage) { ?><div style="display:none;"><?php } ?>
					<div class="thumb-wrapper">
						<div class="thumb">
							<?php echo $thumb_url_start; ?><img src="<?php echo $thumb; ?>" alt="<?php echo $files[$i]; ?>" /><?php echo $thumb_url_end; ?>
						</div> 
						<?php if($show_captions) { ?>
							<div class="caption">
								<?php if($num_chars > $num_captions_chars) { ?>
									<div class="tooltip-container"><?php echo $caption; ?><span class="tooltip"><?php echo $full_caption; ?></span></div>
								<?php } else { echo $caption; } ?>
							</div>
						<?php } ?> 
					</div>
					<?php if($i<$start || $i>=$start + $numPerPage) { ?></div><?php }
				}
			
			} 
			
		}
		
		?> 
	
		<div class="clear"></div>
		  
		<div align="center" class="paginate-wrapper">
			<?php	 
			$urlVars = "album=".urlencode($requested_album)."&amp;";
			echo paginate_array($numPages,$urlVars,$requested_album,$currentPage);
			?>
		</div>
		
		<?php echo (file_exists($album.'/descriptions.txt') ? '<div class="description-wrapper">'.encodeto(itemDescription($album)).'</div>' : ''); //display album description ?>
	 
	<?php	 
	} // end if numFiles not 0	 

}
?>
</div>